jimhigson/oboe.js
# fix unable to run tests on install

@jimhigson 

Fixes #120

This is a fairly large diff (no changes to src files except for formatting), feel free to not review it and just to do the behavior testing below.
 
The short is that I had to update a bunch of dependencies, eventually leading to needing to update jasmine to v2. This version had a some breaking changes, so I had to chase those down. There were a few tests that I couldn't get working. They're skipped for now, but I will file an issue to unskip them. Running the test suite is a little painful now, so I would like to improve that process before going into heavy debugging mode.
 
# How to test

- Use node v4.4.5+
- `$ git clone https://github.com/JuanCaicedo/oboe.js.git`
- `$ npm install` on master. You should see some errors in the console during the install process
- `$ npm test`, which should fail because `ERROR [launcher]: Cannot start PhantomJS`
- `$ rm -r node_modules`
- `$ git reset --hard` (don't worry about this diff, it's a dev set up thing that will be improved later)
- `$ git checkout fix-tests-non-running-on-install`
- `$ npm install`
- `$ npm test`
